src/model/CplaceVersion.ts

```diff
@@ -0,0 +1,98 @@
+import * as fs from 'fs';
+import { cerr, cwarn } from '../utils';
+import * as path from 'path';
+
+export class CplaceVersion {
+    private static readonly VERSION_GRADLE = 'version.gradle';
+
+    private static _version: CplaceVersion | undefined = undefined;
+
+    private constructor(
+        public readonly major: number,
+        public readonly minor: number,
+        public readonly patch: number,
+        public readonly snapshot: boolean
+    ) {}
+
+    public static initialize(currentRepo: string): CplaceVersion {
+        if (CplaceVersion._version !== undefined) {
+            throw new Error(
+                `(CplaceVersion) version has already been initialized`
+            );
+        }
+
+        const versionFilePath = path.resolve(
+            currentRepo,
+            CplaceVersion.VERSION_GRADLE
+        );
+        if (!fs.existsSync(versionFilePath)) {
+            console.warn(
+                cwarn`[NPM] Could not find version.gradle in repo ${currentRepo}...`
+            );
+            console.warn(cwarn`[CplaceVersion] -> Assuming version 1.0.0`);
+            CplaceVersion._version = new CplaceVersion(1, 0, 0, false);
+        } else {
+            const versionFileContent = fs.readFileSync(versionFilePath, 'utf8');
+            const versionString = versionFileContent
+                .split('\n')
+                .find((line) => line.includes('currentVersion'));
+            if (!versionString) {
+                console.error(
+                    cerr`[CplaceVersion] Version string not found in version.gradle file`
+                );
+                throw new Error(
+                    `[CplaceVersion] Version string not found in version.gradle file`
+                );
+            }
+
+            const version = versionString
+                .split('=')[1]
+                .replace(/'/g, '')
+                .trim();
+            const versionSnapshotParts = version.split('-');
+
+            const versionParts = versionSnapshotParts[0].split('.');
+            versionParts.push(
+                versionSnapshotParts.length > 1 &&
+                    versionSnapshotParts[1].toLowerCase().includes('snapshot')
+                    ? 'true'
+                    : 'false'
+            );
+
+            if (versionParts.length < 3) {
+                console.error(
+                    cerr`[CplaceVersion] Expected version to consist of 3 parts`
```

Comment:
For future enhancements: some of our partners use 4 digit versions and other formats. Currently they are definitely not considered to publish artifacts, so this is just to raise the awareness that we might need to adjust this test.